api/client.go

```diff
@@ -0,0 +1,129 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/bitrise-io/go-utils/retry"
+	"github.com/bitrise-io/go-utils/v2/log"
+	"io"
+	"net/http"
+	"net/http/httputil"
+	"time"
+)
+
+const (
+	timeout = 30
+)
+
+type HttpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type DefaultAPIClient struct {
+	httpClient HttpClient
+	authToken  string
+	baseURL    string
+	logger     log.Logger
+}
+
+func NewDefaultAPIClient(baseURL, authToken string, logger log.Logger) DefaultAPIClient {
+	httpClient := retry.NewHTTPClient().StandardClient()
+	httpClient.Timeout = time.Second * timeout
+
+	return DefaultAPIClient{
+		httpClient: httpClient,
+		authToken:  authToken,
+		baseURL:    baseURL,
+		logger:     logger,
+	}
+}
+
+func (c DefaultAPIClient) CreateRelease(appSlug string, params CreateReleaseParameter) (CreateReleaseResponse, error) {
+	req, err := c.request(appSlug, params)
+	if err != nil {
+		return CreateReleaseResponse{}, err
+	}
+
+	dump, err := httputil.DumpRequest(req, false)
+	if err != nil {
+		c.logger.Warnf("request dump failed: %s", err)
+	}
+	c.logger.Debugf("Request dump: %s", string(dump))
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return CreateReleaseResponse{}, err
+	}
+
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			c.logger.Printf(" [!] Failed to close response body: %+v", err)
+		}
+	}()
+
+	dump, err = httputil.DumpResponse(resp, true)
+	if err != nil {
+		c.logger.Warnf("response dump failed: %s", err)
+	}
+	c.logger.Debugf("Response dump: %s\n", string(dump))
+
+	if resp.StatusCode >= 300 || resp.StatusCode < 200 {
+		errResponse, err := c.parseError(resp)
+		if err != nil {
+			return CreateReleaseResponse{}, err
+		}
+
+		return CreateReleaseResponse{}, fmt.Errorf("request to %s has status code %d (should be 2XX): %s", req.URL.String(), resp.StatusCode, errResponse.Message)
+	}
+
+	return c.parseModel(resp)
```

Comment:
NOTE: what about adding a bit more context to the error here? In this case, the network request was successful, likely the release creation is triggered.

Something like this:
```
resp, err := c.parseModel(resp)
if err != nil {
  return CreateReleaseResponse{}, fmt.Errorf("release successfully created but response couldn't be parsed: %s", err)
}
```